### DISCUSSION TITLE
AttributeError: 'UUID' object has no attribute 'translate'

### DISCUSSION THREAD

**Discussion by bert2002** [fastapi-users/fastapi-users#990] [General]
Hi, 
I am moving my code to the new release and run into the following problem. When using the `User` definition it runs into:

```
AttributeError: 'UUID' object has no attribute 'translate'
```
The route I have is:
```
@app.get("/v1/endpoint")
async def get_endpoint(
  user: User = Depends(current_active_verified_user)
  ):

  async with async_session_maker() as conn:
    result = await conn.execute(select(SomeTable).filter(SomeTable.user_id==user.id))

  return result.fetchall()
```
and it runs into the above error. When I remove the filter it works.

The schema for SomeTable is:

```
class SomeTable(Base):
  __tablename__ = 'some_table'
  id = Column(Integer, primary_key=True, autoincrement=True)
  user_id = Column(String(36))
```
Which clearly is a string and not uuid(). Unfortunately MariaDB/Mysql does not support it, but `fastapi-utils` got a `GUID()` option. Its possible to change my user_id to:

```
from fastapi_utils.guid_type import GUID

class SomeTable(Base):
  __tablename__ = 'some_table'
  id = Column(Integer, primary_key=True, autoincrement=True)
  user_id = Column(GUID)
```

With this I am getting rid of the error message, but the responds is an empty `[]` (which it should not). 
My database is still on varchar(36) though:
```
MariaDB [database]> describe some_table;
+-------------+--------------+------+-----+---------+----------------+
| Field       | Type         | Null | Key | Default | Extra          |
+-------------+--------------+------+-----+---------+----------------+
| id          | int(11)      | NO   | PRI | NULL    | auto_increment |
| user_id     | varchar(36)  | YES  |     | NULL    |                |
```
Any idea if the type is the problem or how I call it is the problem?

Many thanks,
bert

**Comment by frankie567** [comment]
I think the problem comes from the fact that we are not using the same GUID implementation as the one in `fastapi-utils`. We use a full `CHAR(36)` implementation, while they use a `CHAR(32)` one. I suggest you use FastAPI Users version to avoid this kind of issue (it's very similar and has proved to work well).

Besides, you also forgot to [configure the foreign key](https://docs.sqlalchemy.org/en/14/orm/basic_relationships.html#basic-relationship-patterns), so MySQL can check that the `user_id` actually exists in the User table. It'll also allow to do convenient things with SQLAlchemy ORM, like relationship loading.

```py
from fastapi_users_db_sqlalchemy.generics import GUID

class SomeTable(Base):
  __tablename__ = 'some_table'
  id = Column(Integer, primary_key=True, autoincrement=True)
  user_id = Column(GUID, ForeignKey("user.id", ondelete="cascade"), nullable=False)
```

**Reply by bert2002** [reply]
I did not know you have a GUID implementation as well :) and yes it works with yours. I will checkout the relationship patterns. Thanks for the hint.
